src/geometry/manifold/vector_bundle/basic.lean

```diff
@@ -0,0 +1,231 @@
+/-
+Copyright (c) 2022 Floris van Doorn, Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn, Heather Macbeth
+-/
+import geometry.manifold.vector_bundle.fiberwise_linear
+import topology.vector_bundle.constructions
+
+/-! # Smooth vector bundles
+
+This file defines smooth vector bundles over a smooth manifold.
+
+Let `E` be a topological vector bundle, with model fibre `F` and base space `B`.  We consider `E` as
+carrying a charted space structure given by its trivializations -- these are charts to `B × F`.
+Then, by "composition", if `B` is itself a charted space over `H` (e.g. a smooth manifold), then `E`
+is also a charted space over `H × F`
+
+Now, we define `smooth_vector_bundle` as the `Prop` of having smooth transition functions.
+Recall the structure groupoid `smooth_fiberwise_linear` on `B × F` consisting of smooth, fibrewise
+linear local homeomorphisms.  We show that our definition of "smooth vector bundle" implies
+`has_groupoid` for this groupoid, and show (by a "composition" of `has_groupoid` instances) that
+this means that a smooth vector bundle is a smooth manifold
+
+Since `smooth_vector_bundle` is a mixin, it should be easy to make variants and for many such
+variants to coexist -- vector bundles can be smooth vector bundles over several different base
+fields, they can also be C^k vector bundles, etc.
+
+## Main definitions and constructions
+
+* `fiber_bundle.charted_space`: A fibre bundle `E` over a base `B` with model fibre `F` is naturally
```

Comment:
It looks strange to me that the Lean name has `fiber_bundle` but the docstring talks of `fibre bundle`. I would uniformize that (or uniformise that, as you like), maybe in a later PR.